pyomo/contrib/doe/documentation/docs/source/index.rst

```diff
@@ -0,0 +1,453 @@
+    >>> measure_variance = {'C': {'CA': 1, 'CB': 1, 'CC': 1}} # provide measurement uncertainty
+    >>> measure_class =  doe.Measurements(measure_pass, variance=measure_variance)  # Use Pyomo.DoE.Measurements to achieve a measurement object
+
+    >>> # === Parameter dictionary ===
+    >>> parameter_dict = {'A1': 84.79, 'A2': 371.72, 'E1': 7.78, 'E2': 15.05}
+
+    >>> # === Define prior information ==
+    >>> prior_none = np.zeros((4,4))
+
+
+Step 3: Compute the FIM of a square MBDoE problem
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This method computes an MBDoE optimization problem with no Degree of Freedom.
+
+.. doctest::
+
+    >>> # === Decide mode ===
+    >>> sensi_opt = 'sequential_finite'
+    >>> # === Specify an experiment ===
+    >>> exp1 = {'CA0': {0: 5}, 'T': {0: 570, 0.125:300,  0.25:300,  0.375:300,  0.5:300,  0.625:300,  0.75:300,  0.875:300, 1:300}}
+    >>> # === Create the DOE object ===
+    >>> doe_object = doe.DesignOfExperiments(parameter_dict, dv_pass, measure_class, create_model,
+    ...                            prior_FIM=prior_none, discretize_model=disc)
+    >>> # === Use ``compute_FIM`` to compute one MBDoE square problem ===
+    >>> result = doe.doe_object.compute_FIM(exp1,mode=sensi_opt, FIM_store_name = 'dynamic.csv',
+    ...                            store_output = 'store_output')
+
+The MBDoE results are stored in this result object. This object can be analyzed as:
+
+.. doctest::
+
+    >>> # === Analyze result ===
+    >>> doe.result.calculate_FIM(doe_object.design_values)
+
+FIM information is then available.
+
+.. doctest::
+
+    >>> result.FIM  # FIM matrix
+    >>> result.design_variable_name # design variable values at each time point
+    >>> result.trace # a scalar number of Trace
+    >>> result.det # a scalar number of determinant
+    >>> result.cond # a scalar number of condition number
+    >>> result.min_eig  # a scalar number of minimal eigen value
+    >>> result.eig # a list of all eigen values
+    >>> result.eigvals # a list of all eigen vectors
+
+
+Step 4: Exploratory analysis (Enumeration)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Exploratory analysis is suggested to enumerate the design space to check if the problem is identifiable, i.e., ensure that D-, E-optimality metrics is not an infinite small number near zero, and Modified E-optimality is not a big number.
+
+Pyomo.DOE accomplishes the exploratory analysis with the ``run_grid_search`` function.
+It allows to define any number of design decisions. Heatmaps can be drawn by two design variables, fixing other design variables.
+1D curve can be drawn by one design variable, fixing all other variables.
+The function ``run_grid_search`` enumerates over the design space, each MBDoE problem accomplished by ``compute_FIM`` method. Therefore, ``run_grid_search`` can also choose from the two modes, ``sequential_finite`` and ``direct_kaug``.
```

Comment:
Therefore, ``run_grid_search`` supports only two modes: ``sequential_finite`` and ``direct_kaug``